docs/user_guide/examples/tutorial_diffusion.ipynb

```diff
@@ -480,17 +506,22 @@
     "    Currently only works for Rectilinear Grids. Operates in place adding a `cell_edge_sizes`\n",
     "    attribute to the grid.\n",
     "    \"\"\"\n",
-    "    assert grid._gtype in (GridType.RectilinearZGrid, GridType.RectilinearSGrid), (\n",
-    "        f\"_cell_edge_sizes() not implemented for {grid._gtype} grids. \"\n",
-    "        \"You can provide cell_edge_sizes yourself by in, e.g., \"\n",
-    "        \"NEMO using the e1u fields etc from the mesh_mask.nc file.\"\n",
-    "    )\n",
-    "\n",
-    "    cell_edge_sizes_x = np.zeros((grid.ydim, grid.xdim), dtype=np.float32)\n",
-    "    cell_edge_sizes_y = np.zeros((grid.ydim, grid.xdim), dtype=np.float32)\n",
-    "\n",
-    "    x_conv = GeographicPolar() if grid.mesh == \"spherical\" else UnitConverter()\n",
-    "    y_conv = Geographic() if grid.mesh == \"spherical\" else UnitConverter()\n",
+    "    # TODO: check for gridtypes once added in v4\n",
+    "    # assert grid._gtype in (GridType.RectilinearZGrid, GridType.RectilinearSGrid), (\n",
+    "    #     f\"_cell_edge_sizes() not implemented for {grid._gtype} grids. \"\n",
+    "    #     \"You can provide cell_edge_sizes yourself by in, e.g., \"\n",
+    "    #     \"NEMO using the e1u fields etc from the mesh_mask.nc file.\"\n",
+    "    # )\n",
+    "\n",
+    "    cell_edge_sizes_x = np.zeros((grid.ydim + 1, grid.xdim + 1), dtype=np.float32)\n",
+    "    cell_edge_sizes_y = np.zeros((grid.ydim + 1, grid.xdim + 1), dtype=np.float32)\n",
+    "\n",
+    "    x_conv = (\n",
+    "        parcels.GeographicPolar()\n",
+    "    )  # if grid._mesh == \"spherical\" else parcels.UnitConverter()\n",
```

Comment:
will `_mesh` ever become a public name so that we can support these types of functions, or should we remove this?